### ISSUE TITLE
Exclude moment.js if not used

### ISSUE THREAD

**Issue by dignifiedquire** [JSxMachina/react-i18nify#28]
I'm only using the `translate` functionality and not the localization, but because of how the bundle is built it includes all of `moment.js` which increases my application size considerably. Currently I have to manually alias `moment` to an empty object in webpack to get rid of it. 

A possible solution could be distributing the original es2015 source, such that webpack can use tree shaking to remove moment when not used.

**Comment by rkrv** [comment]
Big issue for us as well. Another possible solution could be to switch to [date-fns](https://date-fns.org)?

**Comment by gerbenmeyer** [comment]
Solved this now by making `moment.js` an optional peer dependency. Without that installed, everything will work fine, except localizing a date/time.

**Comment by rkrv** [comment]
Hi @gerbenmeyer, thanks a lot for this! Our bundle is significantly smaller now, however, when compiling, I get the following error:

```
./node_modules/react-i18nify/build/lib/I18n.js
Module not found: Can't resolve 'moment' in '/<path_to_project>/node_modules/react-i18nify/build/lib'

./node_modules/react-i18nify/build/lib/I18n.js
Module not found: Can't resolve 'moment/min/locales' in '/<path_to_project>/node_modules/react-i18nify/build/lib'
```

Compilation succeeds, and I do get a bundle, but these warnings are driving me nuts.

**Comment by gerbenmeyer** [comment]
Yeah, I agree we still need a better solution for this.

**Comment by gerbenmeyer** [comment]
Can you check whether you still get this warning with version 1.11.10?

**Comment by rkrv** [comment]
It... somehow made it worse.

```
Failed to compile.

Module not found: Error: Can't resolve 'moment' in '/<path_to_project>/node_modules/react-i18nify/build/lib'


error Command failed with exit code 1.
```

**Comment by gerbenmeyer** [comment]
Strange, because now it's only loaded conditionally. Anyhow, I added an additional `try catch`, hopefully it compiles again. Btw, what tool are you using for compilation?

**Comment by rkrv** [comment]
It does compile now but with warnings.

```
./node_modules/react-i18nify/build/lib/I18n.js
127:8-29 Critical dependency: the request of a dependency is an expression

./node_modules/react-i18nify/build/lib/I18n.js
Module not found: Can't resolve 'moment' in '<project>/node_modules/react-i18nify/build/lib'

./node_modules/react-i18nify/build/lib/I18n.js
Module not found: Can't resolve 'moment/min/locales' in '<project>/node_modules/react-i18nify/build/lib'
```

I am running create-react-app with yarn on osx.

yarn version 1.3.2
create-react-app version 1.4.3

**Comment by gerbenmeyer** [comment]
This will be tackled in the following way: `moment.js` will again be included by default (in `react-i18nify` version 1.11.12 and up), to not break anything. However, the next major version will be without `moment.js`.
